src/Microsoft.Azure.WebJobs.Host/Bindings/Runtime/Binder.cs

```diff
@@ -0,0 +1,172 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Protocols;
+
+namespace Microsoft.Azure.WebJobs.Host.Bindings.Runtime
+{
+    /// <summary>
+    /// Class providing functionality for dynamically binding to WebJobs SDK attributes at runtime.
+    /// </summary>
+    public class Binder : IBinder, IWatchable, IDisposable
```

Comment:
This replaces IBinderEx. It implements all the IBinder capabilities for back compat, but adds the new capabilities needed by Functions. Note that these new capabilities aren't Functions specific - they're actually useful for classic SDK jobs as well.